core/primitives/src/utils.rs

```diff
@@ -180,6 +180,16 @@ pub fn is_valid_sub_account_id(signer_id: &AccountId, sub_account_id: &AccountId
     VALID_ACCOUNT_PART_ID_WITH_TAIL_SEPARATOR.is_match(prefix)
 }
 
+/// Returns a value of the median index of the array, NOT a median value.
+pub fn median<T: std::clone::Clone + std::cmp::Ord>(arr: Vec<T>) -> T {
+    if arr.is_empty() {
+        panic!("empty array")
+    }
+    let mut arr = arr.clone();
```

Comment:
Also, just noticed that you are cloning, but this is not necessary since you have the ownership of this array at the moment. 
You can change the signature of the function to either receive:
`arr: &Vec<T>` or `arr: mut Vec<T>`. If you go with the second case you can get rid of the restriction of the trait `Clone`.

Reply:
Hm, it would be very unexpected that median function changes original array, right?

Reply:
From the signature of the function you have ownership of the vector, so whatever change to it you perform won't affect anyone outside, since you are not returning it.
Whatever vector you pass to this function will be unusable afterwards (ownership is passed in and it is dropped at the end of the function).

 If you choose to pass it by reference, then you should probably create a separate index list and sort that list instead of the of the original vector. It will require that elements implement clone (to clone the answer).

Reply:
Ah, sure, it's ampersand. I was wrong. I will remove cloning.